qa/integration-tests/src/test/java/io/camunda/zeebe/it/network/Ipv6IntegrationTest.java

```diff
@@ -7,150 +7,141 @@
       final Topology topology = client.newTopologyRequest().send().join(5, TimeUnit.SECONDS);
       // then - can find each other
-      TopologyAssert.assertThat(topology).isComplete(3, 1);
+      TopologyAssert.assertThat(topology).isComplete(1, 1);
     }
   }
 
-  private void configureBrokerContainer(final int index, final List<ZeebeBrokerContainer> brokers) {
-    final int clusterSize = brokers.size();
-    final var broker = brokers.get(index);
-    final var hostNameWithoutBraces = getIpv6AddressForIndex(index);
+  private void configureBroker(final ZeebeBrokerNode<?> broker) {
+    final var hostNameWithoutBraces = getIPv6Address(brokerIpAllocator.getAndIncrement());
     final var hostName = String.format("[%s]", hostNameWithoutBraces);
 
     initialContactPoints.add(hostName + ":" + ZeebePort.INTERNAL.getPort());
 
     broker
-        .withNetwork(network)
-        .withNetworkAliases(NETWORK_ALIAS)
-        .withCreateContainerCmdModifier(
-            createContainerCmd ->
-                createContainerCmd
-                    .withIpv6Address(hostNameWithoutBraces)
-                    .withHostName(hostNameWithoutBraces))
-        .withEnv("ZEEBE_BROKER_NETWORK_MAXMESSAGESIZE", "128KB")
-        .withEnv("ZEEBE_BROKER_CLUSTER_NODEID", String.valueOf(index))
-        .withEnv("ZEEBE_BROKER_CLUSTER_CLUSTERSIZE", String.valueOf(clusterSize))
-        .withEnv("ZEEBE_BROKER_CLUSTER_REPLICATIONFACTOR", String.valueOf(REPLICATION_FACTOR))
-        .withEnv("ZEEBE_BROKER_CLUSTER_PARTITIONCOUNT", String.valueOf(PARTITION_COUNT))
+        .withEnv("ZEEBE_LOG_LEVEL", "DEBUG")
+        .withEnv("ATOMIX_LOG_LEVEL", "INFO")
         .withEnv(
             "ZEEBE_BROKER_CLUSTER_INITIALCONTACTPOINTS", String.join(",", initialContactPoints))
-        .withEnv("ZEEBE_BROKER_NETWORK_ADVERTISEDHOST", hostNameWithoutBraces)
-        .withEnv("ZEEBE_LOG_LEVEL", "DEBUG")
+        .withEnv("ZEEBE_BROKER_NETWORK_ADVERTISEDHOST", hostName)
         .withEnv("ZEEBE_BROKER_NETWORK_HOST", INADDR6_ANY)
-        .withEnv("ZEEBE_LOG_LEVEL", "DEBUG")
-        .withEnv("ATOMIX_LOG_LEVEL", "INFO");
+        .withCreateContainerCmdModifier(cmd -> configureHostForIPv6(cmd, hostNameWithoutBraces));
+  }
+
+  private void configureHostForIPv6(
+      final CreateContainerCmd cmd, final String hostNameWithoutBraces) {
+    final var hostConfig = Optional.ofNullable(cmd.getHostConfig()).orElse(new HostConfig());
+    cmd.withHostConfig(hostConfig.withNetworkMode(network.getId()));
+    cmd.withIpv6Address(hostNameWithoutBraces).withHostName(hostNameWithoutBraces);
   }
 
-  private void configureGatewayContainer(
-      final ZeebeGatewayContainer gateway, final String initialContactPoint) {
-    final String address = getIpv6AddressForIndex(CLUSTER_SIZE);
+  private void configureGateway(final ZeebeGatewayNode<?> gateway) {
+    final String contactPoint = getIPv6Address(3);
+    final String hostNameWithoutBraces = getIPv6Address(2);
```

Comment:
:wrench: This makes it really hard to understand. Could we just define these addresses statically at the top of the class? We have anyway only a gateway and a broker right so wouldn't be that much?

Reply:
Yeah, good call, I was being lazy and just blindly refactoring what was there :+1: